packages/docusaurus-theme-common/src/utils/useAlternatePageUtils.ts

```diff
@@ -43,15 +45,21 @@ export function useAlternatePageUtils(): {
   function createUrl({
     locale,
     fullyQualified,
+    queryString,
   }: {
     locale: string;
     // For hreflang SEO headers, we need it to be fully qualified (full protocol/domain/path...)
     // For locale dropdown, using a path is good enough
     fullyQualified: boolean;
+    // Appends a query string parameter with the selected locale
+    queryString?: string;
   }) {
+    const queryStringSuffix = queryString
+      ? `?${queryString}=${encodeURIComponent(locale)}`
```

Comment:
Why do you need `?foo=pt-BR`, can't `?persistLocale=true` be good enough?


`queryString` config implicitly means you are passing a whole querystring as config (`x=y`), not a querystring attribute name (`x`).

We'd rather rename this `queryString` config to something more meaningful (like `localeQueryStringParam: 'locale'` ?), or pass the whole querystring object as is without unexpected extra logic